src/Type/Doctrine/Descriptors/ReflectionDescriptor.php

```diff
@@ -57,6 +69,20 @@ public function getWritableToDatabaseType(): Type
 
 	public function getDatabaseInternalType(): Type
 	{
+		$registry = $this->container->getByType(DefaultDescriptorRegistry::class);
+		$parents = $this->reflectionProvider->getClass($this->type)->getParentClassesNames();
```

Comment:
Better have a hasClass() guard here

Reply:
Lets agree if you want https://github.com/phpstan/phpstan-doctrine/pull/582#discussion_r1652239416 or not first.

Reply:
Added